charts/rasa-x/templates/_helpers.tpl

```diff
@@ -210,3 +219,53 @@ Return 'true' if an enterprise version is run.
 {{- print "false" -}}
 {{- end -}}
 {{- end -}}
+
+{{/*
+Return the rasa-x.version value as a default if the dbMigrationService.tag variable is not defined.
+*/}}
+{{- define "db-migration-service.version" -}}
+{{ .Values.dbMigrationService.tag | default (include "rasa-x.version" .) }}
+{{- end -}}
+
+{{/*
+Return 'true' if required version to run the database migration service is correct.
+If version is not valid semantic version then not use the DB migration service.
+*/}}
+{{- define "db-migration-service.requiredVersion" -}}
+{{- if and (not (regexMatch "^(v[0-9]|0|[1-9]\\d*).(0|[1-9]\\d*)\\.(0|[1-9]\\d*)(?:-((?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\\.(?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\\+([0-9a-zA-Z-]+(?:\\.[0-9a-zA-Z-]+)*))?$" (include "db-migration-service.version" .))) (not .Values.dbMigrationService.ignoreVersionCheck) -}}
```

Comment:
> ^(v[0-9]|0|[1-9]

Isn't this enough `^(v[0-9]+.` ? (similar for some of the following).

 And do we actually need such a check? `semverCompare` will fail anyway if it's not a valid version, no?

Reply:
Actually, the regex is copy & paste from the https://semver.org/ website, so I assume it's the best one to check semantic version 😄   The check purpose is to deal with a scenario where someone uses a custom non-sematic tag for a Rasa X image, instead of throwing an error that a version is not valid, we assume that the version is not the required version to run the DB migration service.  Wdyt?

Reply:
Makes sense! Wouldn't it be better to fail loud and early instead of silently disabling it?

Reply:
I was a bit driven by the https://github.com/RasaHQ/rasa-x-helm/issues/86 issue, but you're right. It's better to fail loudly. Additionally, there is the `dbMigrationService.ignoreVersionCheck` parameter that can be used to disable the check. Thanks 🙌

Reply:
I'll do a PR for the #86 later this week 👍